src/shared/telemetry/telemetryEvent.ts

```diff
@@ -5,39 +5,58 @@
 
 'use strict'
 
-import { MetadataEntry, MetricDatum, Unit } from './clienttelemetry'
+import { AwsContext } from '../awsContext'
+import { MetadataEntry, MetricDatum } from './clienttelemetry'
+import { Datum, TelemetryNamespace } from './telemetryUtils'
 
 const NAME_ILLEGAL_CHARS_REGEX = new RegExp('[^\\w+-.:]', 'g')
 const REMOVE_UNDERSCORES_REGEX = new RegExp('_', 'g')
 
-export interface Datum {
-    name: string
-    value: number
-    unit?: Unit
-    metadata?: Map<string, string>
-}
-
 export interface TelemetryEvent {
     namespace: string
     createTime: Date
     data?: Datum[]
 }
 
-export function toMetricData(array: TelemetryEvent[]): MetricDatum[] {
+export function toMetricData(
+    array: TelemetryEvent[],
+    trimmedAwsContext: Pick<AwsContext, 'getCredentialAccountId'>
+): MetricDatum[] {
     return ([] as MetricDatum[]).concat(
         ...array.map( metricEvent => {
 
             const namespace = metricEvent.namespace.replace(REMOVE_UNDERSCORES_REGEX, '')
+            let metadata: MetadataEntry[] = []
+            const account = trimmedAwsContext.getCredentialAccountId()
+
+            if (namespace === TelemetryNamespace.Session) {
+                // this matches JetBrains' functionality: the AWS account ID is not set on session start.
+                metadata.push({
+                    Key: 'account',
+                    Value: undefined
+                })
+            } else {
+                if (account) {
+                    metadata.push({
+                        Key: 'account',
+                        Value: account
```

Comment:
You could collapse this if/else to use
`Value: account || ''`
Or you could make a non exported method
`getAccountIdOrDefault(trimmedAwsContext, '')`